libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -190,7 +190,23 @@ vector<Declaration const*> NameAndTypeResolver::nameFromCurrentScope(ASTString c
 Declaration const* NameAndTypeResolver::pathFromCurrentScope(vector<ASTString> const& _path) const
 {
 	solAssert(!_path.empty(), "");
-	vector<Declaration const*> candidates = m_currentScope->resolveName(_path.front(), true);
+	vector<Declaration const*> candidates;
+
+	bool isStructScope = dynamic_cast<StructDefinition const*>(m_currentScope->currentNode());
```

Comment:
What would be the enclosing node here?

Reply:
For the following contract
```
contract C {
    enum E {a}
    struct S {E X; uint E;}
}
```
if `isStructScope` is `true`, it would be `ContractDefinition`.